src/cloud-api-adaptor/podvm/qcow2/rhel/qemu-rhel.pkr.hcl

```diff
@@ -14,14 +14,15 @@ source "qemu" "rhel" {
   iso_checksum     = "${var.cloud_image_checksum}"
   iso_url          = "${var.cloud_image_url}"
   output_directory = "output"
-  qemuargs         = [["-m", "${var.memory}"], ["-smp", "cpus=${var.cpus}"], ["-cdrom", "${var.cloud_init_image}"], ["-serial", "mon:stdio"], ["-cpu", "Cascadelake-Server"]]
+  qemuargs         = [["-m", "${var.memory}"], ["-smp", "cpus=${var.cpus}"], ["-cdrom", "${var.cloud_init_image}"], ["-serial", "mon:stdio"], ["-cpu", "${var.cpu_type}"]]
```

Comment:
nit, if this cpu types are usually constants for the architectures,  you can do something like the following instead:
`cpu_model    = "${var.os_arch}" == "x86_64" ? "Cascadelake-Server" : "max"`

Reply:
I tried it but seems like there is a failure. I also tried passing os_arch = s390x as a variable, seems no luck